File: solutions_5751500831719424_0/C++/ABHI1234/repeat.cpp
#include<iostream>
#include<cmath>
using namespace std;
int main(){
  int t,l=1,n,i,j,g,k,m;
  cin>>t;
  int ans;
  string a[2];
  while(l<=t){
    cin>>n;
    cin>>a[0]>>a[1];
    i = 0;g = 0;
    ans = 0;
    m = a[0].length();
    n = a[1].length();
    while(i < m || g < n ){
      j =0;k=0;
      while(i+j<m && a[0][i] == a[0][i+j] ){
	j++;
      }
      while(g + k < n && a[1][g] == a[1][g+k] ){
	k++;
      }
      if(a[1][g] != a[0][i]){
    	cout<<"Case #"<<l<<": Fegla Won\n";
	 i = m;
	 g = n;
	 ans = -1;
	}
     else{
       if(j != k)
        ans = ans + abs(j-k);
       i += j;
       g += k;
     }
    }
    if (ans != -1)
    cout<<"Case #"<<l<<": "<<ans<<endl;
    l++;
  }
  return 0;
}

File: solutions_5751500831719424_0/Python/superoven/repeat.py
def gl(f, splitchar=' '):
    return map(f, raw_input().split(splitchar))

def g(f):
    return f(raw_input())

def all(strs):
    for i in xrange(len(strs)-1):
        if strs[i] != strs[i+1]:
            return False
    return True
            

def fil(inp):
    curr=''
    out=''
    for l in inp:
        if l == curr:
            continue
        curr = l
        out += l
    return out

def num(inp):
    out =[]
    count = 1
    curr =inp[0]
    for i in xrange(1,len(inp)):
        if inp[i] != curr:
            out.append(count)
            count = 1
            curr = inp[i]
            continue
        count += 1
        curr = inp[i]
    out.append(count)
    return out

t=g(int)
for ind in xrange(t):
    n=g(int)
    strs=[g(str) for _ in xrange(n)]
    ans = map(lambda x: fil(x), strs)
    print "Case #%d:"%(ind+1),
    if not all(ans):
        print "Fegla Won"
        continue
    nums=map(lambda x: num(x), strs)
    count = 0
    for i in xrange(len(nums[0])):
        vals=sorted(map(lambda x: x[i], nums))
        med=vals[len(vals)/2]
        add = sum(map(lambda x: abs(x-med), vals))
        count += add
    print count
